mlflow/utils/databricks_utils.py

```diff
@@ -489,11 +501,36 @@ def get_model_dependency_oauth_token(should_retry=True):
             ) from e
 
 
+class TrackingURIConfigProvider(DatabricksConfigProvider):
+    """
+    TrackingURIConfigProvider extracts `scope` and `key_prefix` from tracking URI
+    of format like `databricks://scope:key_prefix`,
```

Comment:
1. Are we expect this tracking URI format specified by users? If so, is there any document we can link about the format?
2. What happens if user doesn't have `{key_prefix}-host` secret?
3. I found this line in bug-bash example `mlflow.set_tracking_uri("databricks://E2-DOG")`, is this valid input to this provider?

Reply:
> Are we expect this tracking URI format specified by users?

Yes. We have related doc in `get_databricks_host_creds`. But this should not be a recommended way. We recommend to use environmental variable or ~/.databrickscfg to config credential info.


`databricks://scope:key_prefix` 
it is different with format `databricks://E2-DOG` (this is `databricks://<profile-name>` format),

> What happens if user doesn't have {key_prefix}-host secret?
then this provider can't find valid credential info.

Reply:
But, `databricks://scope:key_prefix` is a deprecated way.

Note that `databricks://scope:key_prefix` way can only be used in **databricks runtime** (it uses `dbutils`), but , in databricks runtime, MLflow can automatically get dynamic token value from dbutils (See `DynamicConfigProvider`)
so setting tracking uri to `databricks://scope:key_prefix` is meaningless, it is only used for keeping compatible with legacy code

Reply:
Ah makes sense, thank you for the explanation!! (maybe a quick deprecation comment in `TrackingURIConfigProvider` is helpful for new ppl like me)